### PR TITLE
chore: Add jest testing framework

### PR DESCRIPTION
**Pull request by rhysbarrett**
This commit adds `jest` as the testing, mocking and assertion framework. It also adds a simple test to verify setup.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,13 +22,19 @@
...
+    "@types/supertest": "^2.0.10",
     "copyfiles": "^2.3.0",
     "gulp": "^4.0.2",
     "gulp-sass": "^4.1.0",
+    "http-status-codes": "^1.4.0",
+    "jest": "^26.4.0",
     "node-sass": "^4.14.1",
     "nodemon": "^2.0.4",
+    "supertest": "^4.0.2",
+    "ts-jest": "^26.2.0",
```
Comment (akaJSaunders) [inline package.json:37]: have you made sure these are the latest version available?

Comment (rhysbarrett) [reply package.json:37]: I installed via npm rather than by copying from Appeals, so I'd hope so. I'll double check.

Comment (rhysbarrett) [reply package.json:37]: Confirmed, these are all latest versions.

Comment (akaJSaunders) [reply package.json:37]: Good job!

File: jest.config.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+module.exports = {
+    roots: [
+        "<rootDir>"
+    ],
+    testPathIgnorePatterns: [
+        "/node_modules/",
+        "/dist/"
+    ],
+    transform: {
+        "^.+\\.(ts|tsx)$": "ts-jest"
```
Comment (akaJSaunders) [inline jest.config.js:10]: what is this regex doing?

File: jest.config.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+module.exports = {
+    roots: [
+        "<rootDir>"
```
Comment (akaJSaunders) [inline jest.config.js:3]: is this rootDir set within our application?

File: jest.config.js (hunk 2 above)
Comment (rhysbarrett) [reply jest.config.js:10]: Added this on advice from the tutorial I was following, something to do with the fact we're compiling from Typescript to JavaScript. However I'm able to run the tests without it, so I've removed it.

File: jest.config.js (hunk 3 above)
Comment (rhysbarrett) [reply jest.config.js:3]: Doesn't need to be, it's a predefined token within Jest

Comment (rhysbarrett) [reply jest.config.js:3]: Having said that, according to the documentation this is the default value for the attribute, so it doesn't need to be there. I'll remove it.

File: jest.config.js (hunk 2 above)
Comment (rhysbarrett) [reply jest.config.js:10]: Cancel that, I was editing the wrong repository (the temporary one I was using before you set up the dev environment). The tests break when it's removed, so it appears necessary.

**Pull request merged**